### PR TITLE
Port regionWorker, step 3

### PR DESCRIPTION
**Pull request by xiaobogaga**
@coocood PTAL, thanks.

### CODE REVIEW

**Comment by xiaobogaga** [comment]
@coocood  PTAL, thanks.

File: tikv/raftstore/worker.go [hunk 1]
```diff
@@ -290,7 +291,75 @@ type pendingDeleteRanges struct {
...
+
+func (peerInfo *stalePeerInfo) serializeStalePeerInfo() []byte {
+	if timeoutData, err := peerInfo.timeout.MarshalBinary(); err != nil {
+		log.Errorf("serialize stale Peer info failed, [regionId: %d, endKey: %v]", peerInfo.regionId, peerInfo.endKey)
+		return nil
+	} else {
+		buf := make([]byte, 8+4+4)
+		binary.LittleEndian.PutUint64(buf, peerInfo.regionId)
+		binary.LittleEndian.PutUint32(buf[8:], uint32(len(peerInfo.endKey)))
+		binary.LittleEndian.PutUint32(buf[12:], uint32(len(timeoutData)))
```
Comment (coocood) [inline tikv/raftstore/worker.go]: We can just store a single int64 for time.
```
        t := time.Now()
	n := t.UnixNano()
	t = time.Unix(0, n)
```
And we also don't need to store the length of endkey, if the total length of value is known, and other two fields takes 16 bytes.

File: lockstore/lockstore.go [hunk 2]
```diff
@@ -20,6 +20,7 @@ type MemStore struct {
 
 	// We only consume 2 bits for a random height call.
 	rand rand.Source64
+	size int
```
Comment (coocood) [inline lockstore/lockstore.go]: Use `length` is better.

File: tikv/raftstore/worker.go [hunk 3]
```diff
@@ -290,7 +291,75 @@ type pendingDeleteRanges struct {
...
+				startKey: startKey,
+				endKey:   peerInfo.endKey,
+				regionId: peerInfo.regionId,
+			})
+		}
+	}
+	return
+}
+
+type stalePeerInfo struct {
```
Comment (coocood) [inline tikv/raftstore/worker.go]: You can define this type like this
```
type stalePeerInfo struct {
    data []byte
}
```
Then define methods to get those fields.
So we don't need to call serialize and deserialize.

**Comment by xiaobogaga** [comment]
@coocood  PTAL again, thanks.

**Comment by coocood** [comment]
LGTM

**Pull request merged**